### PR TITLE
 REST: search with FieldCriterion

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
New feature based on:

As a Developer I want to search using REST FieldCriterion
https://jira.ez.no/browse/EZP-28783
https://github.com/ezsystems/ezpublish-kernel/pull/2239

As a Developer I want SQL Search by Field to use ezkeyword Ext. Storage
https://jira.ez.no/browse/EZP-28843
https://github.com/ezsystems/ezpublish-kernel/pull/2256

### CODE REVIEW

**Comment by MagdalenaZuba** [comment]
ping @ViniTou for review

File: docs/guide/search.md [hunk 1]
```diff
@@ -422,6 +422,59 @@ ezpublish.search.legacy.gateway.sort_clause_handler.location.depth:
...
+    
+### Search using custom Field Criterion [REST]
+      
+REST search can be performed via `POST /views` using custom `FieldCriterion`. This allows you to build custom content logic queries with nested logical operators OR/AND/NOT. This functionality works for all custom fields. 
+
+Custom Field Criterion search mirrors the one already existing in PHP API `eZ\Publish\API\Repository\Values\Content\Query\Criterion\Field` by exposing it to REST.
+      
+##### Example of custom Content Query:
+      
+```php
```
Comment (ViniTou) [inline docs/guide/search.md]: this is not php, it's json, dunno if it matters in markdown.

File: docs/guide/search.md [hunk 2]
```diff
@@ -230,7 +230,7 @@ A Criterion consist of two parts (similar to Sort Clause and Facet Builder):
 - The API Value: `Criterion`
 - Specific handler per search engine: `CriterionHandler`
 
-`Criterion` represents the value you use in the API, while `CriterionHandler` deals with the business logic in the background translating the value to something the search engine can understand.
+`Criterion` represents the value you use in the API, while `CriterionHandler` deals with the business logic in the background translating the value to something the search engine can understand. FieldValue `CriterionHandler` also handles ezkeyword External Storage for Legacy (SQL-based) search.
```
Comment (DominikaK) [inline docs/guide/search.md]: Maybe "Field value" and "`ezkeyword`"?

Comment (ViniTou) [inline docs/guide/search.md]: >FieldValue `CriterionHandler` also handles ezkeyword External Storage for Legacy (SQL-based) search.

tbh, I don't have a single clue what that could mean.

File: docs/guide/search.md [hunk 3]
```diff
@@ -422,6 +422,59 @@ ezpublish.search.legacy.gateway.sort_clause_handler.location.depth:
 !!! note "See also"
 
     See also [Symfony documentation about Service Container](http://symfony.com/doc/current/book/service_container.html#service-parameters) for passing parameters.
+    
+### Search using custom Field Criterion [REST]
+      
+REST search can be performed via `POST /views` using custom `FieldCriterion`. This allows you to build custom content logic queries with nested logical operators OR/AND/NOT. This functionality works for all custom fields. 
+
```
Comment (ViniTou) [inline docs/guide/search.md]: >This functionality works for all custom fields. 

Don't know if this is the case, but this search don't works for all the fields (like rich text for example.)

File: docs/guide/search.md (hunk 2 above)
Comment (ViniTou) [reply docs/guide/search.md]: Remove FieldValue from this sentence.

**Pull request merged**